CppCoreGuidelines.md

```diff
@@ -14924,8 +14924,38 @@ There is no explicit locking and both correct (value) return and error (exceptio
 
 ##### Example
 
-    ???
+    class Moonphase_error : public std::exception
+    {};
+
+    int fun(int i) noexcept
+    {
+        return i + 2;
+    }
+
+    int fun_ex()
+    {
+        // ...
+        throw Moonphase_error{};
+        // ...
+    }
+
+    void async_example()
+    {
+        auto f = std::async(std::launch::async, fun, 10);
 
+        std::cout << f.get() << '\n';
+
+        try
+        {
+            auto fe = std::async(std::launch::async, fun_ex);
+            auto i = fe.get(); // this call will thow a Moonphase_error exception
```

Comment:
s/thow/throw
but also, this example seems to take a lot of real estate to demo custom error handling, but shows no concurrent execution. How about something more like `parallel_sum` from  https://en.cppreference.com/w/cpp/thread/async#Example

Reply:
ok, thank you. I see your point.
Give me a little bit to think about it.

Reply:
I have committed another example.
Please let me know what you think about it.